model/textparse/interface.go

```diff
@@ -112,16 +113,4 @@ const (
 	EntryHistogram Entry = 5 // A series with a native histogram as a value.
 )
 
-// MetricType represents metric type values.
-type MetricType string
-
-const (
-	MetricTypeCounter        = MetricType("counter")
-	MetricTypeGauge          = MetricType("gauge")
-	MetricTypeHistogram      = MetricType("histogram")
-	MetricTypeGaugeHistogram = MetricType("gaugehistogram")
-	MetricTypeSummary        = MetricType("summary")
-	MetricTypeInfo           = MetricType("info")
-	MetricTypeStateset       = MetricType("stateset")
-	MetricTypeUnknown        = MetricType("unknown")
-)
+type MetricType = model.MetricType
```

Comment:
I'm curious about this is still needed here 🤔 

I can see `
func (c *scrapeCache) setType(metric []byte, t textparse.MetricType) {` in scrape/scrape.go line 961, but wouldn't it make sense to just change the signature there to model.MetricType too?

Reply:
I think it might due to many other importers (e.g. Mimir, Cortex, Thanos) who would have bad times if we braking change migration at once (: Perhaps @bboreham tries to smooth this out, which is great.

Reply:
Aaaah good point. Maybe worth adding a deprecation note to allow linter to flag this to downstream projects? Then we could eventually remove the variable

Reply:
Removing this provokes another 7 changes in Prometheus, which aren't material to this PR.  That's the main reason I included it.
And as you say downstream projects can be affected, though they will also be affected by the move of constants.

I'm under the impression that deprecating it will force an error from our linter.

Maybe just rip off the bandaid and make the change?

Reply:
I made this change.